crates/yttrium/src/pay/mod.rs

```diff
@@ -22,72 +33,831 @@ impl<T: std::fmt::Debug> From<progenitor_client::Error<T>> for PayError {
+        let response = with_retry(|| async {
+            with_sdk_config!(
+                self.client.gateway_get_payment_options(),
+                &self.config
+            )
+            .id(&payment_id)
+            .body(body.clone())
+            .send()
+            .await
+        })
+        .await
+        .map_err(map_payment_options_error)?;
+
+        let api_response = response.into_inner();
+
+        // Cache the options with their required actions
+        let cached: Vec<CachedPaymentOption> = api_response
+            .options
+            .iter()
+            .map(|o| CachedPaymentOption {
+                option_id: o.id.clone(),
+                required_actions: o
+                    .required_actions
+                    .iter()
+                    .cloned()
+                    .map(Into::into)
+                    .collect(),
+            })
+            .collect();
+        let mut cache = self.cached_options.write().map_err(|e| {
+            GetPaymentOptionsError::InternalError(format!(
+                "Cache lock poisoned: {}",
+                e
+            ))
+        })?;
+        *cache = cached;
+
+        Ok(api_response.into())
+    }
+
+    /// Get required payment actions for a selected option
+    /// Returns the list of actions from the cached options (must call get_payment_options first)
+    pub fn get_required_payment_actions(
+        &self,
+        option_id: String,
+    ) -> Result<Vec<RequiredAction>, GetPaymentRequestError> {
+        let cache = self.cached_options.read().map_err(|e| {
+            GetPaymentRequestError::InternalError(format!(
+                "Cache lock poisoned: {}",
+                e
+            ))
+        })?;
+
+        let cached_option = cache
+            .iter()
+            .find(|o| o.option_id == option_id)
+            .ok_or_else(|| {
+                GetPaymentRequestError::OptionNotFound(format!(
+                    "Option {} not found in cache. Call get_payment_options first.",
+                    option_id
+                ))
+            })?;
+
+        Ok(cached_option.required_actions.clone())
+    }
+
+    /// Fetch an action for a payment option
+    pub async fn fetch(
+        &self,
+        payment_id: String,
+        option_id: String,
+        data: String,
+    ) -> Result<FetchResponse, PayError> {
+        let data_value: serde_json::Value =
+            serde_json::from_str(&data).unwrap_or(serde_json::Value::Null);
+        let body = types::FetchRequest { option_id, data: data_value };
+        let response = with_retry(|| async {
+            with_sdk_config!(self.client.fetch_handler(), &self.config)
+                .id(&payment_id)
+                .body(body.clone())
+                .send()
+                .await
+        })
+        .await?;
+        Ok(response.into_inner().into())
+    }
+
+    /// Get gateway payment status (for polling)
+    async fn get_gateway_payment_status(
```

Comment:
Do we need it in a public interface?
It's already called inside `confirm_payment` when the status of the payment is not final

Reply:
its not public, don't have `pub` modifier